src/main/scala/com/actian/spark_vector/sql/SparkSqlTable.scala

```diff
@@ -0,0 +1,30 @@
+package com.actian.spark_vector.sql
+
+import java.util.concurrent.atomic.AtomicLong
+import org.apache.spark.sql.DataFrame
+
+sealed trait SparkSqlTable {
+  def tableName: String
+  def quotedName = sparkQuote(tableName)
+  def close(): Unit
+}
+
+case class HiveTable(override val tableName: String) extends SparkSqlTable {
+  override def close(): Unit = { }
+}
+
+class TempTable private (override val tableName: String, df: DataFrame) extends SparkSqlTable {
```

Comment:
The idea was to disallow the creation of unregistered TempTables. To be consistent, I now made register() private.